src/Signer.php

```diff
@@ -29,33 +29,31 @@ public function getAlgorithmId(): string;
 
     /**
      * Apply changes on headers according with algorithm
-     *
-     * @param array $headers
      */
-    public function modifyHeader(array &$headers);
+    public function modifyHeader(array $headers): array;
 
     /**
-     * Returns a signature for given data
+     * Creates a hash for the given payload
      *
      * @param string $payload
-     * @param Key|string $key
+     * @param Key $key
      *
-     * @return Signature
+     * @return string
      *
      * @throws InvalidArgumentException When given key is invalid
      */
-    public function sign(string $payload, $key): Signature;
+    public function sign(string $payload, Key $key): string;
```

Comment:
@lcobucci can this branch be merged into that one? I need to see the bigger picture here, because this change is a regression (worse API, from my PoV)